src/objects/agent.py

```diff
@@ -1,10 +1,33 @@
-import operator
-
-from .object import Object
+from src.actions.move import Move
 from src.common.enums import Direction
 from src.common.point import Point
+from .object import Object
 
 
 class Agent(Object):
-    def __init__(self, position: Point = Point(0, 0)):
-        super().__init__(position)
+    '''
+    Simple agent existing in a `scene`.
+    '''
+
+    def __init__(self, scene, position: Point = Point(0, 0)):
+        super().__init__(position, scene)
+
+    def tick(self) -> None:
+        '''
+        Called on an `Object`, so it can perform some actions
+        '''
+        agent_builtins = {
+            'move': lambda direction: Move(self, direction).execute(self.scene),
+            'Direction': Direction,
+            'len': len,
+            'range': range,
+            'enumerate': enumerate,
+        }
+
+        # Sure, I know exec bad
+        # pylint: disable=exec-used
+        exec(
+            self.scene.runtime.agents_code,  # It violates lasagna code rule but I guess its ok
```

Comment:
If it violates then let's don't do it
It's gonna become big, so making it messy now will make it really hard later
let's move `agents_code` to Agent
Agent shouldn't have the access to runtime

Reply:
Will see if it works, but I'm not a huge fan of passing an object from input to runtime to scene to agent.